architecture/ambient/peer-authentication.md

```diff
@@ -0,0 +1,68 @@
+# PeerAuthentication Implementation in Ambient
```

Comment:
May be worth adding a quick note that this is the implementation description - how things work - but not necessarily how they should work. 

PERMISSIVE in istio-sidecar is quite different in behavior and semantics from PERMISSIVE in ambient.

There is also a 'DISABLED' mode which should be described ( "Connection is not tunneled." is the 
description in the docs), as well as the auto-mtls which is intended to replace PeerAuthentication .

If I remember correctly the decision ( at some point in the past ) was to not promote PeerAuthentication to v1 because of those problems - may be worth including this in the doc too.

Reply:
Good catch on DISABLED. In Ambient, implementation-wise, it's effectively the same as PERMISSIVE (i.e. plaintext can be used). I'm not sure I'm following you re: auto-mtls; my understanding is that PeerAuthentication is a knob for determining if plaintext is allowed for a given application. Auto-mtls is just our ALPN hack to detect whether something is in the mesh or not and fall back